lib/postprocessing/image.js

```diff
@@ -94,113 +93,50 @@ function processBasicImageOperations(img, instructions) {
     await logImageMagickVersion();
 
-    return new Promise(function (resolve, reject) {
-        const outdir = path.dirname(outfile);
-        const renditionName = path.basename(outfile);
-
-        // for pdfs and multi-page docs/images, take the first one only
-        let img = gm(`${infile}[0]`);
-
-        // const selectedOperations = operations || DEFAULT_OPERATIONS;
-        // turning off watermarking for increamental post-processing
-        const selectedOperations = DEFAULT_OPERATIONS;
-        img = applyImageOperations(img, instructions, selectedOperations);
-
-        img.write(outfile,
-            function (err) {
-                if (err) {
-                    console.log("FAILURE of rendition processing using image processing lib (imagemagick)", renditionName);
-                    console.error("failed conversion:", err);
-                    reject(`conversion using image processing lib (imagemagick) failed: ${err.message}, code: ${err.code}, signal: ${err.signal}`);
-                } else {
-                    // If we specified jpegSize for JPEG output we need to verify that the size does not exceed that
-                    const size = fileSize(outfile);
-                    if (instructions.fmt === "jpg" && instructions.jpegSize && size > instructions.jpegSize) {
-                        const newImg = gm(outfile);
-
-                        // From http://www.imagemagick.org/Usage/formats/#jpg
-                        // As of IM v6.5.8-2 you can specify a maximum output filesize for the JPEG image.
-                        newImg.define(`jpeg:extent=${instructions.jpegSize}`);
-                        const tmpFile = `${outdir}/${tempFileName("jpg")}`;
-                        newImg.write(tmpFile,
-                            function (err) {
-                                if (err) {
-                                    console.log("FAILURE of rendition processing (fmt=jpg) using image processing lib (imagemagick)", renditionName);
-                                    console.error("failed conversion:", err);
-                                    reject(`cconversion using image processing lib (imagemagick) failed: ${err.message}, code: ${err.code}, signal: ${err.signal}`);
-                                } else {
-                                    fs.renameSync(tmpFile, outfile);
-                                    console.log("END of rendition processing using image processing lib (imagemagick)", renditionName);
-                                    resolve(true);
-                                }
-                            });
-                    } else {
-                        console.log("END of rendition processing using image processing lib (imagemagick)", renditionName);
-                        resolve(true);
-                    }
-                }
-            });
-    });
+    const img = await read(infile);
+
+    await render(img, instructions);
```

Comment:
Would also be nicer, I think, to be able to chain the calls (fluent API mentionned above).

Reply:
Same comment as above:
1. for these 3 lines (read, render, write) there is no benefit in fluent. Fluent api is good if you have an API that is used _many_ times differently, setting various configs, for short pieces of code that represent specific configs/operations. But we don't have that here, since we have a generic bridge from our instructions API to the imagemagick/gm API. In such cases fluent doesn't bring anything to the table.
2. assuming we really wanted to have a fluent API, having it on `img` doens't help, we'd have to turn `render()` and `write()` into new classes with a fluent API, since these are own functions that are not part of `img` (`gm` API).

Reply:
`Fluent api is good if you have an API that is used many times differently, setting various configs, for short pieces of code that represent specific configs/operations` <= that is what I expect to start happening when post-processing is in sdk. I think some people will look into our code and start calling this directly to do image processing. Hence my suggestion for the api to be nice and fluent, even if for now it is a private one. It won't hurt doing it - also what is the point of having `img` as const? If it is for performance gains they're minimal.

Reply:
We don't offer the `gm` or any other javascript API to custom workers here. The only API is the rendition instructions.

If a custom worker wants to do something more complex with imagemagick (once they get access to it), they can use the `gm` library themselves in their code. But that would be separate from our (internal) post processing code.